src/setup.jl

```diff
@@ -112,24 +111,30 @@ function regenerate_gaproot()
     haskey(ENV, "GAP_CXX") && pushfirst!(cxx_candidates, ENV["GAP_CXX"])
     CXX = sysinfo["GAP_CXX"] = select_compiler("C++", cxx_candidates, ".cc")
 
-    sysinfo["GAP_CFLAGS"] = " -g -O2"
-    sysinfo["GAP_CXXFLAGS"] = " -g -O2"
+    ##sysinfo["GAP_CFLAGS"] = " -g -O2"
+    ##sysinfo["GAP_CXXFLAGS"] = " -g -O2"
+    #sysinfo["GAP_CFLAGS"] = " -pthread -g -O2"
+    #sysinfo["GAP_CXXFLAGS"] = " -pthread -g -O2"
 
     # set include flags
     gap_include = joinpath(gap_prefix, "include", "gap")
     gap_include2 = joinpath(gaproot_mutable) # for code doing `#include "src/compiled.h"`
     sysinfo["GAP_CPPFLAGS"] = "-I$(gap_include) -I$(gap_include2) -DUSE_JULIA_GC=1"
 
-    # set linker flags; since these are meant for use for GAP packages, add the necessary
-    # flags to link against libgap
+    # set linker flags; since these are meant for use for GAP packages,
+    # add the necessary flags to link against libgap
     gap_lib = joinpath(gap_prefix, "lib")
     sysinfo["GAP_LDFLAGS"] = "-L$(gap_lib) -lgap"
 
-    # adjust linker flags for GAP kernel extensions on macOS
+    # set gac compiler flags (get appended to GAP_CFLAGS and GAP_CXXFLAGS)
+    sysinfo["GAC_CFLAGS"] = "-fPIC"
     if Sys.isapple()
-        sysinfo["GAC_LDFLAGS"] = "-bundle -L$(gap_lib) -lgap"
+        sysinfo["GAC_CFLAGS"] = "-fno-common"
     end
```

Comment:
perhaps
`sysinfo["GAC_CFLAGS"] = Sys.isapple() ? "-fno-common" : "--fPIC"`?

Reply:
In fact we can omit it now as the file provided by `GAP_jll` already got this right :-). I've pushed a commit cleaning this and a few other flags up, please have a look.

One thing confuses me, though: 
```
julia> gap_prefix = GAP.GAP_jll.find_artifact_dir();

julia> sysinfo = GAP.Setup.read_sysinfo_gap(joinpath(gap_prefix, "lib", "gap"));

julia> haskey(sysinfo,"GAP_CPPFLAGS")
false

julia> haskey(GAP.sysinfo,"GAP_CPPFLAGS")
true
```

So the file we get from `GAP_jll` does not define `GAP_CPPFLAGS`. But I am pretty sure GAP 4.13.0 sets `GAP_CPPFLAGS` in there... Huh. Anyway, we (still) need to override it, I am just confused by this discrepency. Ah well.

Reply:
The line in question in the file from GAP 4.13.0 is as follows.
```
GAP_CPPFLAGS="-I/workspace/destdir/include/gap -I/workspace/destdir/include -DUSE_JULIA_GC=1"
```
It contains two `=` characters.
Thus the condition from `read_sysinfo_gap` is violated that only lines are considered which contain exactly one `=`.

Reply:
great catch, thanks Thomas